lib/bundler/cli/add.rb

```diff
@@ -2,34 +2,38 @@
 
 module Bundler
   class CLI::Add
+    attr_reader :gems, :options, :version
+
     def initialize(options, gems)
       @gems = gems
       @options = options
-      @options[:group] = @options[:group].split(",").map(&:strip) if !@options[:group].nil? && !@options[:group].empty?
+      @options[:group] = options[:group].split(",").map(&:strip) if !options[:group].nil? && !options[:group].empty?
+      @version = options[:version].split(",").map(&:strip) unless options[:version].nil?
     end
 
     def run
-      raise InvalidOption, "You can not specify `--strict` and `--optimistic` at the same time." if @options[:strict] && @options[:optimistic]
+      validate_options!
```

Comment:
I moved all validations to `validate_options!` I added the exclamation mark because it can raise an exception.